sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -93,6 +93,23 @@ class FilesElement extends Polymer.Element {
    * to initialize element state.
    */
   ready() {
+    // Get the last startup path. Must be called before super.ready() to beat the
+    // currentPath changed event.
+    ApiManager.getUserSettings()
+      .then((settings: UserSettings) => {
+        if (settings.startuppath) {
+          let path = settings.startuppath;
+          // For backward compatibility with the current path format.
+          if (path.startsWith('/tree/')) {
+            path = path.substr('/tree/'.length);
+          }
+          if (path.startsWith(this.basePath)) {
```

Comment:
Shouldn't the check for basePath go before the check for /tree/? If I understand basePath correctly, you would get something like /base-path/tree/ in the path coming in. Or is /tree/ no longer used when there is a basePath?